tests/trace_plugin_test.py

```diff
@@ -0,0 +1,85 @@
+#!/usr/bin/env python3
+import time
+import unittest
+
+from testUtils import Utils
+from Cluster import Cluster
+from TestHelper import TestHelper
+from Node import Node
+from WalletMgr import WalletMgr
+from core_symbol import CORE_SYMBOL
+
+class TraceApiPluginTest(unittest.TestCase):
+    sleep_s = 1
+    cluster=Cluster(walletd=True, defproduceraPrvtKey=None)
+    walletMgr=WalletMgr(True)
+    accounts = []
+    account_names = ["alice", "bob", "charlie"]
+    cluster.setWalletMgr(walletMgr)
+
+    # kill nodeos and keosd and clean up dir
+    def cleanEnv(self) :
+        self.cluster.killall(allInstances=True)
+        self.cluster.cleanup()
+        self.walletMgr.killall(allInstances=True)
+        self.walletMgr.cleanup()
+
+    # start keosd and nodeos
+    def startEnv(self) :
+        self.cluster.launch(totalNodes=1, enableTrace=True)
+        self.walletMgr.launch()
+        testWalletName="testwallet"
+        testWallet=self.walletMgr.create(testWalletName, [self.cluster.eosioAccount, self.cluster.defproduceraAccount])
+        self.cluster.validateAccounts(None)
+        self.accounts=Cluster.createAccountKeys(len(self.account_names))
+        node = self.cluster.getNode(0)
+        for idx in range(len(self.account_names)):
+            self.accounts[idx].name =  self.account_names[idx]
+            self.walletMgr.importKey(self.accounts[idx], testWallet)
+        for account in self.accounts:
+            node.createInitializeAccount(account, self.cluster.eosioAccount, buyRAM=1000000, stakedDeposit=5000000, waitForTransBlock=True, exitOnError=True)
+        time.sleep(self.sleep_s)
+
+    def test_TraceApi(self) :
+        node = self.cluster.getNode(0)
+        for account in self.accounts:
+            self.assertIsNotNone(node.verifyAccount(account))
+
+        expectedAmount = Node.currencyIntToStr(5000000, CORE_SYMBOL)
+        account_balances = []
+        for account in self.accounts:
+            amount = node.getAccountEosBalanceStr(account.name)
+            self.assertEqual(amount, expectedAmount)
+            account_balances.append(amount)
+
+        xferAmount = Node.currencyIntToStr(123456, CORE_SYMBOL)
+        trans = node.transferFunds(self.accounts[0], self.accounts[1], xferAmount, "test transfer a->b")
+        transId = Node.getTransId(trans)
+        blockNum = Node.getTransBlockNum(trans)
+
+        self.assertEqual(node.getAccountEosBalanceStr(self.accounts[0].name), Utils.deduceAmount(expectedAmount, xferAmount))
+        self.assertEqual(node.getAccountEosBalanceStr(self.accounts[1].name), Utils.addAmount(expectedAmount, xferAmount))
+        time.sleep(self.sleep_s)
+        base_cmd_str = ("curl http://%s:%s/v1/") % (TestHelper.LOCAL_HOST, node.port)
```

Comment:
make a method Node.get_trace_api_block, that uses Node.processCurlCmd so you just need to pass in the block num here, and if we add more trace_api testing, we can reuse this.

Reply:
You mean Node.py, right? It is very generic function, better to make it specific for specific plugin. But I can create a local function.